mava/systems/ppo/sebulba/rec_mappo.py

```diff
@@ -0,0 +1,858 @@
+    init_x = (init_obs, init_done)
+
+    # Initialise hidden states.
+    init_policy_hstate = ScannedRNN.initialize_carry(
+        (config.arch.num_envs, config.system.num_agents), config.network.hidden_state_dim
+    )
+    init_critic_hstate = ScannedRNN.initialize_carry(
+        (config.arch.num_envs, config.system.num_agents), config.network.hidden_state_dim
+    )
+
+    # Initialise params and optimiser state.
+    actor_params = actor_network.init(actor_key, init_policy_hstate, init_x)
+    actor_opt_state = actor_optim.init(actor_params)
+    critic_params = critic_network.init(critic_key, init_critic_hstate, init_x)
+    critic_opt_state = critic_optim.init(critic_params)
+
+    # Pack params and initial states.
+    params = Params(actor_params, critic_params)
+    hstates = HiddenStates(init_policy_hstate, init_critic_hstate)
+
+    # Pack apply and update functions.
+    apply_fns = (actor_network.apply, critic_network.apply)
+    update_fns = (actor_optim.update, critic_optim.update)
+
+    # Defines how the learner state is sharded: params, opt and key = replicated, timestep = sharded
+    learn_state_spec = RNNLearnerState(
+        model_spec, model_spec, data_spec, None, data_spec, data_spec, data_spec
+    )
+
+    learn = get_learner_step_fn(apply_fns, update_fns, config)
+    learn = jax.jit(
+        shard_map(
+            learn,
+            mesh=mesh,
+            in_specs=(learn_state_spec, data_spec),
+            out_specs=(learn_state_spec, data_spec),
+        )
+    )
+
+    # Load model from checkpoint if specified.
+    if config.logger.checkpointing.load_model:
+        loaded_checkpoint = Checkpointer(
+            model_name=config.logger.system_name,
+            **config.logger.checkpointing.load_args,  # Other checkpoint args
+        )
+        # Restore the learner state from the checkpoint
+        restored_params, _ = loaded_checkpoint.restore_params(input_params=params)
+        # Update the params
+        params = restored_params
+
+    # Define params to be replicated across devices and batches.
+    key, *step_keys = jax.random.split(key, len(learner_devices) + 1)
+    step_keys = jnp.stack(step_keys, 0)
+
+    opt_states = OptStates(actor_opt_state, critic_opt_state)
+    dones = jnp.zeros(
+        (config.arch.num_envs, config.system.num_agents),
+        dtype=bool,
+    )
+
+    # Duplicate learner across Learner devices.
+    params, opt_states, hstates, step_keys, dones = jax.device_put(
+        (params, opt_states, hstates, step_keys, dones), learner_sharding
+    )
+
+    # Initialise learner state.
+    init_learner_state = RNNLearnerState(params, opt_states, step_keys, None, None, dones, None)  # type: ignore
+    env.close()
+
+    return learn, apply_fns, init_learner_state, learner_sharding  # type: ignore
+
+
+def run_experiment(_config: DictConfig) -> float:
+    """Runs experiment."""
+    _config.logger.system_name = "rec_mappo_sebulba.yaml"
```

Comment:
system name shouldn't have a file extension
```suggestion
    _config.logger.system_name = "rec_mappo_sebulba"
```